cmsBuild

```diff
@@ -355,14 +355,38 @@ def downloadGit(source, dest, options):
     return False
   return packCheckout(args["tempdir"], args["dest"], args["export"])
 
+def downloadPip(source, dest, options):
+
+# pip:/package/version/tarbalname
+  sp=source.split('/')
+  pack=sp[2].strip()
+  if len(sp)>4: pack=pack+'=='+sp[3].strip()
+  tarball=sp[-1]
+#newer pip will need this (but untested)
+#  comm = 'which pip; pip download --no-deps ' + dest + ' ' + pack
+  comm = 'which pip; pip install --no-deps -d ' + dest + ' ' + pack+'; /bin/ls '+dest+'| grep -v files.list > '+dest+'/files.list'
+  error, output = getstatusoutput(comm)
+  if error:
+    log("Error while downloading sources from %s using pip.\n\n %s \n %s \n  %s \n " % (source, comm, error, output) )
+    return False
+
+  comm2='cd '+dest+'; tar cfz '+tarball+' files.list `cat files.list`'
+  error, output = getstatusoutput(comm2)
+  if error:
+    log("Error while downloading sources from %s using pip.\n\n %s \n %s \n  %s \n " % (source, comm, error, output) )
+    return False
+
```

Comment:
I personally would download it in temp directory and then move the final source.tar.gz to the dest.